src/kzg/mod.rs

```diff
@@ -48,7 +48,8 @@ impl KzgSettings {
         };
 
         let roots_of_unity_brp = {
-            let mut reversed_roots = roots_of_unity.clone();
+            let mut reversed_roots = Box::new([Fr::one(); FIELD_ELEMENTS_PER_BLOB]);
+            reversed_roots.copy_from_slice(&roots_of_unity);
```

Comment:
You can get rid of this complexity by creating roots directly as a Box on line 39.

Reply:
i think i need to keep it as we get a `SIGSEGV: invalid memory reference` when creating the roots as a box and then cloning them here